prime-router/src/main/kotlin/fhirengine/utils/HL7Reader.kt

```diff
@@ -87,18 +195,23 @@ class HL7Reader(private val actionLogger: ActionLogger) : Logging {
          * @return the timestamp or null if not specified
          */
         fun getMessageTimestamp(message: Message): Date? {
-            val timestamp = (message["MSH"] as MSH).msh7_DateTimeOfMessage
-            return if (!timestamp.isEmpty && !timestamp.ts1_Time.isEmpty) {
-                timestamp.ts1_Time.valueAsDate
-            } else null
+            return when (val structure = message["MSH"]) {
+                is ca.uhn.hl7v2.model.v27.segment.MSH -> structure.msh7_DateTimeOfMessage.valueAsDate
+                is ca.uhn.hl7v2.model.v251.segment.MSH -> structure.msh7_DateTimeOfMessage.ts1_Time.valueAsDate
```

Comment:
Optional but we could import these segments with things like `import ca.uhn.hl7v2.model.v251.segment.MSH as v251-MSH` similarly to my suggestion for the message models.